### PR TITLE
Webpack

### PR DESCRIPTION
**Pull request by jpmedley**
@addyosmani @gauntface @jeffposnick

### CODE REVIEW

File: src/get-started/webpack.md [hunk 1]
```diff
@@ -1,8 +1,45 @@
...
-title: Get Started > Webpack
+title: Get Started > webpack
 ---
 
-# Webpack
+# webpack
 
-Need to write up.
+The following example uses [workbox-webpack-plugin](https://www.npmjs.com/package/workbox-webpack-plugin)
+to create a precaching service worker in your webpack build process. This will
```
Comment (gauntface) [inline src/get-started/webpack.md]: I'm curious if "precaching service worker" is a common enough term or not. I would have gone down the path of

"The following examples uses the workbox-webpack-plugin to generate a service worker file capable of precaching your site for offline use."

"Each time you run your web pack build, the service worker will be updated for the latest file changes."

File: src/get-started/webpack.md [hunk 2]
```diff
@@ -1,8 +1,45 @@
...
+give you a new service worker every time you run it.
+
+1. [Install Node.js](https://nodejs.org/en/).
```
Comment (gauntface) [inline src/get-started/webpack.md]: If you are using webpack this is a given.

File: src/get-started/webpack.md [hunk 3]
```diff
@@ -1,8 +1,45 @@
...
+
+    ```
+    npm install workbox-webpack-plugin --save-dev
+    ```
+
+1. After installing the webpack plugin add the following to your `webpack.config.js`.
+
+    ```
+    const path = require('path');
+    const WorkboxBuildWebpackPlugin = require('workbox-webpack-plugin');
```
Comment (gauntface) [inline src/get-started/webpack.md]: Change "WorkboxBuildWebpackPlugin" to "workboxPlugin"

File: src/get-started/webpack.md [hunk 4]
```diff
@@ -1,8 +1,45 @@
...
+    const DIST_DIR = 'dist';
+
+    module.exports = { /* Do the usual webpack stuff. */
+      entry: './app/index.js',
+      output: {
+        filename: 'bundle.js',
+        path: path.resolve(__dirname, DIST_DIR),
+      },
+      plugins: [ /* Call the plugin. */
+        new WorkboxBuildWebpackPlugin({
```
Comment (gauntface) [inline src/get-started/webpack.md]: Change "WorkboxBuildWebpackPlugin" to "workboxPlugin"

File: src/get-started/webpack.md [hunk 5]
```diff
@@ -1,8 +1,45 @@
...
+          globDirectory: DIST_DIR,
+          staticFileGlobs: ['**/*.{html,js,css}'],
+          swDest: path.join(DIST_DIR, 'sw.js'),
+        }),
+      ]
+    };
+    ```
+
+   **Note:** The plugin for generating the service worker should always be
+   run as the last step in each build. This ensures that your service worker
```
Comment (gauntface) [inline src/get-started/webpack.md]: Comments from npm scripts apply here as well.

Comment (gauntface) [reply src/get-started/webpack.md]: May be worth exploring add this as part of a getting started layout (i.e. a layout that adds this at the end of every page using that layout). Ping me if you think it's be useful.

File: src/get-started/webpack.md (hunk 1 above)
Comment (addyosmani) [reply src/get-started/webpack.md]: 👍 Matt's phrasing "The following examples uses the workbox-webpack-plugin to generate a service worker file capable of precaching your site for offline use."

File: src/get-started/webpack.md (hunk 2 above)
Comment (addyosmani) [reply src/get-started/webpack.md]: It's useful to look at how other Webpack plugins handle the install flow: https://github.com/jantimon/html-webpack-plugin/. They assume both Node and Webpack are already installed and dive straight-into adding in the plugin.

File: src/get-started/webpack.md (hunk 5 above)
Comment (jpmedley) [reply src/get-started/webpack.md]: I'd rather not, frankly.

**Comment by jpmedley** [comment]
@gauntface PTAL

**Review by addyosmani — APPROVED** [review]

**Pull request merged**